src/runtime-rs/crates/runtimes/src/shim_mgmt/handlers.rs

```diff
@@ -146,3 +148,19 @@ async fn direct_volume_resize_handler(
         _ => Err(anyhow!("handler: Failed to resize volume")),
     }
 }
+
+// returns the url for metrics
+async fn metrics_url_handler(
+    sandbox: Arc<dyn Sandbox>,
+    _req: Request<Body>,
+) -> Result<Response<Body>> {
+    // get metrics from agent, hypervisor, and shim
+    let agent_metrics = sandbox.agent_metrics().await.unwrap_or_default();
+    let hypervisor_metrics = sandbox.hypervisor_metrics().await.unwrap_or_default();
+    let shim_metrics = get_shim_metrics().unwrap_or_default();
+
+    Ok(Response::new(Body::from(format!(
+        "{}{}{}",
```

Comment:
Is it needed to add `\n` between these three types of metrics to join them as a string?

Reply:
Since `runtime-go` does not add `\n` between these three types of metrics, I followed this style.

https://github.com/kata-containers/kata-containers/blob/afbc1f94d795e60ef772e4d7e13e65c24b5d00f2/src/runtime/pkg/containerd-shim-v2/shim_management.go#L82-L111